File: bindings/javascript/Polycode/ShaderPlatformData.js
function ShaderPlatformData() {
	Object.defineProperties(this, {
		'shaderID': { enumerable: true, configurable: true, get: ShaderPlatformData.prototype.__get_shaderID, set: ShaderPlatformData.prototype.__set_shaderID},
		'vertexProgramID': { enumerable: true, configurable: true, get: ShaderPlatformData.prototype.__get_vertexProgramID, set: ShaderPlatformData.prototype.__set_vertexProgramID},
		'fragmentProgramID': { enumerable: true, configurable: true, get: ShaderPlatformData.prototype.__get_fragmentProgramID, set: ShaderPlatformData.prototype.__set_fragmentProgramID}
	})
}
ShaderPlatformData.prototype.__get_shaderID = function() {
	var retVal = new GLuint()
	retVal.__ptr = 	Polycode.ShaderPlatformData__get_shaderID(this.__ptr)
	return retVal
}

ShaderPlatformData.prototype.__set_shaderID = function(val) {
	Polycode.ShaderPlatformData__set_shaderID(this.__ptr, val.__ptr)
}

ShaderPlatformData.prototype.__get_vertexProgramID = function() {
	var retVal = new GLuint()
	retVal.__ptr = 	Polycode.ShaderPlatformData__get_vertexProgramID(this.__ptr)
	return retVal
}

ShaderPlatformData.prototype.__set_vertexProgramID = function(val) {
	Polycode.ShaderPlatformData__set_vertexProgramID(this.__ptr, val.__ptr)
}

ShaderPlatformData.prototype.__get_fragmentProgramID = function() {
	var retVal = new GLuint()
	retVal.__ptr = 	Polycode.ShaderPlatformData__get_fragmentProgramID(this.__ptr)
	return retVal
}

ShaderPlatformData.prototype.__set_fragmentProgramID = function(val) {
	Polycode.ShaderPlatformData__set_fragmentProgramID(this.__ptr, val.__ptr)
}

Duktape.fin(ShaderPlatformData.prototype, function (x) {
	if (x === ShaderPlatformData.prototype) {
		return;
	}
	Polycode.ShaderPlatformData__delete(x.__ptr)
})

File: bindings/lua/Polycode/ShaderPlatformData.lua
class "ShaderPlatformData"


function ShaderPlatformData:__getvar(name)
	if name == "shaderID" then
		local retVal = Polycode.ShaderPlatformData_get_shaderID(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["GLuint"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "vertexProgramID" then
		local retVal = Polycode.ShaderPlatformData_get_vertexProgramID(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["GLuint"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "fragmentProgramID" then
		local retVal = Polycode.ShaderPlatformData_get_fragmentProgramID(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["GLuint"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function ShaderPlatformData:__setvar(name,value)
	if name == "shaderID" then
		Polycode.ShaderPlatformData_set_shaderID(self.__ptr, value.__ptr)
		return true
	elseif name == "vertexProgramID" then
		Polycode.ShaderPlatformData_set_vertexProgramID(self.__ptr, value.__ptr)
		return true
	elseif name == "fragmentProgramID" then
		Polycode.ShaderPlatformData_set_fragmentProgramID(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function ShaderPlatformData:__delete()
	if self then Polycode.delete_ShaderPlatformData(self.__ptr) end
end
